python/ray/__init__.py

```diff
@@ -255,6 +258,21 @@ def __getattr__(self, attr):
     "PlacementGroupID",
 ]
 
+if sys.version_info < (3, 7):
+    # TODO(Clark): Remove this one we drop Python 3.6 support.
+    from ray import autoscaler  # noqa: F401
+    from ray import data  # noqa: F401
+    from ray import workflow  # noqa: F401
+else:
+    # Delay importing of expensive, isolated subpackages.
+    def __getattr__(name: str):
+        import importlib
+
+        if name in _subpackages:
+            return importlib.import_module("." + name, __name__)
+        raise AttributeError(f"module {__name__!r} has no attribute {name!r}")
```

Comment:
Can we add a test for this behavior?

Reply:
Hmm this one might be interesting to test... I suppose we can check that `module.__getattribute__` returns the expected module but errors for `__getattr__` for the top-level subpackages that we define and are eagerly imported?